ablySpec/RealtimeClientChannel.swift

```diff
@@ -607,6 +607,55 @@ class RealtimeClientChannel: QuickSpec {
                     expect(Test.counter).toEventually(equal(2), timeout: testTimeout)
                 }
 
+                context("message cannot be decoded or decrypted") {
+
+                    // RTL7e
+                    it("should deliver with encoding attribute set indicating the residual encoding and error should be emitted") {
+                        let options = AblyTests.commonAppSetup()
+                        options.autoConnect = false
+                        let client = ARTRealtime(options: options)
+                        client.setTransportClass(TestProxyTransport.self)
+                        client.connect()
+                        defer { client.close() }
+
+                        let channelOptions = ARTChannelOptions(encrypted: ARTCrypto.defaultParams())
```

Comment:
> and TZ3a says that if the key isn't passed it is generated randomly.

Shit, you're right. I now recall why `TZ3a` does that as the spec was largely based on what was implemented in Java, although I don't really like the fact that it does that. We should leave in the spec, but it's a weird default behaviour if you ask me.

In regards to IV, it feels a bit like an extension of the `key` issue we have with `defaultParams()`, why are we auto-generating an `IV` randomly that both sides will need to synchronise on.  The `IV` should either be shared between clients before communicating, or both sides should have a means to generate an `IV` and keep it in sync.  The latter option is clearly more complicated, but equally far more robust in the long term.

I have not documented Crypto yet, and was intentionally leaving this to later as I didn't think it was high priority, but I expect I will struggle a bit if we're not all entirely clear on this.

@paddybyers is this worth a quick chat to discuss?